test/replication-luatest/on_replication_split_brain_rollback_event_test.lua

```diff
@@ -0,0 +1,496 @@
+                            local type = stmt.op_type
+                            local arg
+                            if request_type == box.iproto.type.REPLACE or
+                               request_type == box.iproto.type.INSERT or
+                               request_type == box.iproto.type.UPSERT then
+                                arg = stmt.tuple
+                                t.assert_equals(stmt.key, nil)
+                            else
+                                arg = stmt.key
+                            end
+                            local ops
+                            if request_type == box.iproto.type.UPDATE then
+                                ops = stmt.tuple
+                                t.assert_equals(stmt.ops, nil)
+                            else
+                                ops = stmt.ops
+                            end
+                            local old = stmt.old_tuple
+                            local new = stmt.new_tuple
+
+                            t.assert_equals(term, box.info.synchro.queue.term)
+                            t.assert_equals(rid, box.info.id)
+                            if counter == n_stmts then
+                               t.assert_equals(lsn, box.info.lsn)
+                            end
+                            t.assert_equals(sid, box.space.s.id)
+                            t.assert_equals(type, request_type)
+                            t.assert_equals(arg, request_arg)
+                            if request_ops ~= nil then
+                                t.assert_equals(ops, request_ops)
+                            end
+                            if counter == 1 then
+                                t.assert_equals(old, old_tuple)
+                            else
+                                t.assert_equals(old, arg)
+                            end
+                            t.assert_equals(new, new_tuple)
+                            counter = counter + 1
+                        end)
+                    end
+                    t.assert_equals(n_stmts, counter - 1)
+                end
+            end)
+        box.cfg{replication_synchro_quorum = ''}
+        box.space.s:replace{777, 0}
+        box.cfg{replication_synchro_quorum = 3}
+    end)
+end)
+
+-- Test that the event iterator returns correct information for different
+-- transaction statements.
+g.test_different_stmts = function(cg)
+    local stmts = {
+        {'replace', {0}, nil, nil, {0}},
+        {'replace', {777, 1}, nil, {777, 0}, {777, 1}},
+        {'insert', {0}, nil, nil, {0}},
+        {'delete', {0}, nil, nil, nil},
+        {'delete', {777}, nil, {777, 0}, nil},
+        {'update', {777}, {{'=', 2, 1}}, {777, 0}, {777, 1}},
+        {'update', {777}, {{'=', 2, 1}, {'=', 3, 1}}, {777, 0}, {777, 1, 1}},
+        {'upsert', {777}, {{'=', 2, 1}}, {777, 0}, {777, 1}},
+        {'upsert', {0}, {{'=', 2, 1}}, nil, {0}},
+    }
```

Comment:
Have you thought about maybe having them as key-value tables? Like
```
local stmts = {
    {type = 'replace', key = ..., tuple = ... },
    ...
}
```
Tbh, it is quite hard to read at the moment, and you have to pass all those items individually in the code above, while could pass each statement as a key-value table. I feel like this would simplify the code above quite a bit.

Reply:
Thanks, applied your suggestion, it indeed looks better now.

I guess the lesson of this patch series for me is to stop passing so much stuff around 😁